packages/opentelemetry-exporter-collector/test/common/transform.test.ts

```diff
@@ -63,62 +63,6 @@ describe('transform', () => {
         { key: 'foo', value: { doubleValue: 1.34 } },
       ]);
     });
-    it('should convert attribute mixed with maps and array', () => {
```

Comment:
I think you have  removed quite important test, you should just convert the input to have homogenous arrays for example
```
const attributes: Attributes = {
  foo1: [1,2,3],
  foo2: ['a','b','c'],
  foo3: {
    arr1: [true, false],
    val1: true,
    val2: 'true',
    val3: null,
    val4: undefined,
    foo4: {
      a: 1,
      b: true,
      c: null
    }
  }
};
```
will be a valid list of attributes

Reply:
> I think you have removed quite important test, you should just convert the input to have homogenous arrays for example
> 
> ```
> const attributes: Attributes = {
>   foo1: [1,2,3],
>   foo2: ['a','b','c'],
>   foo3: {
>     arr1: [true, false],
>     val1: true,
>     val2: 'true',
>     val3: null,
>     val4: undefined,
>     foo4: {
>       a: 1,
>       b: true,
>       c: null
>     }
>   }
> };
> ```
> 
> will be a valid list of attributes

I removed it because arbitrarily nested values are not valid according to the spec.

>The attribute key, which MUST be a non-null and non-empty string.
>The attribute value, which is either:
>- A primitive type: string, boolean, double precision floating point (IEEE 754-1985) or signed 64 bit integer.
>- An array of primitive type values. The array MUST be homogeneous, i.e. it MUST NOT contain values of different types. For protocols that do not natively support array values such values SHOULD be represented as JSON strings.

Reply:
You right, the spec issue -> https://github.com/open-telemetry/opentelemetry-specification/issues/376 seems like this is still opened and have been postponed even though the proto already implemented that (https://github.com/open-telemetry/opentelemetry-proto/pull/157).